### PR TITLE
[SYCL][ESIMD] More updates to print performance data in uniform way

### PR DESCRIPTION
**Pull request by sndmitriev**
This patch changes stencil, linear, mandelbrot, BitonicSort and
Prefix_local_sum tests to print performance data the same way as it
was done in PR#524 for histogram tests.

Signed-off-by: Sergey Dmitriev <serguei.n.dmitriev@intel.com>

### CODE REVIEW

File: SYCL/ESIMD/Prefix_Local_sum2.cpp [hunk 1]
```diff
@@ -122,53 +121,74 @@ int main(int argc, char *argv[]) {
...
 
-  unsigned sum[2];
-  sum[0] = sum[1] = 0;
-  for (int i = 0; i < 256; i++) {
-    sum[0] += pInputs[i * TUPLE_SZ];
-    sum[1] += pInputs[i * TUPLE_SZ + 1];
-  }
+  // allocate device buffer
+  unsigned int *pDeviceOutputs = static_cast<unsigned int *>(
+      malloc_shared(size * TUPLE_SZ * sizeof(unsigned int), dev, ctxt));
```
Comment (v-klochkov) [inline SYCL/ESIMD/Prefix_Local_sum2.cpp]:
```suggestion
  unsigned int *pDeviceOutputs = malloc_shared<unsigned int>(size * TUPLE_SZ, q));
```

File: SYCL/ESIMD/Prefix_Local_sum2.cpp [hunk 2]
```diff
@@ -122,53 +121,74 @@ int main(int argc, char *argv[]) {
...
+  esimd_test::display_timing_stats(kernel_times, num_iters,
+                                   (end - start) * 1000);
+
+  bool pass = memcmp(pDeviceOutputs, pExpectOutputs,
                      size * TUPLE_SZ * sizeof(unsigned int)) == 0;
   std::cout << "Prefix " << (pass ? "=> PASSED" : "=> FAILED") << std::endl
             << std::endl;
 
-  free(pInputs, ctxt);
+  free(pDeviceOutputs, ctxt);
```
Comment (v-klochkov) [inline SYCL/ESIMD/Prefix_Local_sum2.cpp]: After this change here and in few other places, 'ctxt' may be removed, I think.
```suggestion
  free(pDeviceOutputs, q);
```

**Review by v-klochkov — DISMISSED** [review]
Looks good to me. 

I noticed that malloc_shared calls may be written more compact way and also if use free(ptr, q) instead of 'free(ptr, ctxt)', then 'ctxt' can be removed. This is a non-blocking comment.

File: SYCL/ESIMD/Prefix_Local_sum2.cpp (hunk 2 above)
Comment (sndmitriev) [reply SYCL/ESIMD/Prefix_Local_sum2.cpp]: Done.

**Review by v-klochkov — APPROVED** [review]

**Review by kbobrovs — APPROVED** [review]

**Pull request merged**